File: src/main/java/com/saiph/application/GestionRH/Domain/dto/FormationDto.java
package com.saiph.application.GestionRH.Domain.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.saiph.application.GestionRH.Domain.entities.User;
import jakarta.persistence.ManyToMany;
import lombok.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
@Getter
@Setter
public class FormationDto extends GenericDto {


    @NonNull
        @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dateD;
    @NonNull
        @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dateF;
    @NonNull
    private Integer duree;
    @NonNull
    private String lieu;
    @NonNull
    private Integer nbrPlace;
    @NonNull
    private String img;
    @NonNull
    private String titre;
    @NonNull
    private String description;

    @ManyToMany
    private List<User> ListEmploye = new ArrayList<>();
}

File: Gestion-Rh-UI/src/app/services/models/formation-dto.ts
/* tslint:disable */
/* eslint-disable */
import { User } from '../models/user';
export interface FormationDto {
  createdBy?: string;
  createdDate?: string;
  dateD?: string;
  dateF?: string;
  description?: string;
  duree?: number;
  id?: number;
  img?: string;
  lastModifiedBy?: string;
  lastModifiedDate?: string;
  lieu?: string;
  listEmploye?: Array<User>;
  nbrPlace?: number;
  titre?: string;
}
